source/mir/random/flex/internal/area.d

```diff
@@ -0,0 +1,668 @@
+module mir.random.flex.internal.area;
+
+import mir.random.flex.internal.types : Interval;
+import std.traits : ReturnType;
+
+/**
+Determines the hat and squeeze function of an interval.
+Based on Theorem 1 of Botts et al. (2013).
+*/
+void determineSqueezeAndHat(S)(ref Interval!S iv)
+{
+    import mir.utility.linearfun : linearFun;
+    import mir.random.flex.internal.types : determineType, FunType;
+
+    // y (aka x0) is defined to be the maximal point of the boundary points
+    enum sec = `(iv.ltx >= iv.rtx) ?
+                linearFun!S((iv.rtx - iv.ltx) / (iv.rx - iv.lx), iv.lx, iv.ltx) :
+                linearFun!S((iv.rtx - iv.ltx) / (iv.rx - iv.lx), iv.rx, iv.rtx)`;
+
+    enum t_l = "linearFun!S(iv.lt1x, iv.lx, iv.ltx)";
+    enum t_r = "linearFun!S(iv.rt1x, iv.rx, iv.rtx)";
+
+    // could potentially be saved for subsequent calls
+    FunType type = determineType(iv);
+    with(FunType) with(iv)
+    switch(type)
+    {
+        case T1a:
+            squeeze = mixin(t_r);
+            hat = mixin(t_l);
+            break;
+        case T1b:
+            squeeze = mixin(t_l);
+            hat = mixin(t_r);
+            break;
+        case T2a:
+            squeeze = mixin(sec);
+            hat = mixin(t_l);
+            break;
+        case T2b:
+            squeeze = mixin(sec);
+            hat = mixin(t_r);
+            break;
+        case T3a:
+            squeeze = mixin(t_r);
+            hat = mixin(sec);
+            break;
+        case T3b:
+            squeeze = mixin(t_l);
+            hat = mixin(sec);
+            break;
+        case T4a:
+            if (iv.lx == -S.infinity)
+            {
+                squeeze = squeeze.init;
+                hat = mixin(t_r);
+                break;
+            }
+            if (iv.rx == +S.infinity)
+            {
+                squeeze = squeeze.init;
+                hat = mixin(t_l);
+                break;
+            }
```

Comment:
I still think that these `if` special cases should have comments clarifying exactly what part of the Tinflex paper they are described in.

Also, picky, but can we be certain that both `if` conditions won't apply?  It seems to be an implicit assumption of the code here.